data/transactions/logic/eval.go

```diff
@@ -159,6 +162,8 @@ type LedgerForLogic interface {
 	DelGlobal(key string) error
 
 	GetDelta(txn *transactions.Transaction) (evalDelta basics.EvalDelta, err error)
+
+	AppendLog(value basics.TealValue) error
```

Comment:
Why take the argument as a `TealValue` instead of `[]byte`?

Reply:
I used TealValue to be consistent with existing methods like SetLocal/SetGlobal

Reply:
Those methods only use `TealValue` for global/local state values, which can be either integers or bytes. They actually use `string` for the keys which are bytes-only, so perhaps `AppendLog` should also use `string` to be consistent with them.